doc/api/fs.md

```diff
@@ -5902,6 +5934,33 @@ this API: [`fs.mkdtemp()`][].
 The optional `options` argument can be a string specifying an encoding, or an
 object with an `encoding` property specifying the character encoding to use.
 
+### `fs.mkdtempDisposableSync(prefix[, options])`
+
+<!-- YAML
+added: TODO
+changes:
+  - version:
+      - TODO
+    pr-url: TODO
+    description: Initial version.  <-- this doesn't seem to be common practice but the result is that there is no pr-url for the initial feature
+-->
+
+* `prefix` {string|Buffer|URL}
+* `options` {string|Object}
+  * `encoding` {string} **Default:** `'utf8'`
+* Returns: {object} A disposable object with a "path" property.
+
+Returns a disposable object whose `path` property holds the created directory
+path. When the object is disposed, the directory and its contents will be
+removed. The object has a `remove()` method which will perform the same task.
+
+<!-- TODO: link MDN docs for disposables once https://github.com/mdn/content/pull/38027 lands -->
+
+For detailed information, see the documentation of [`fs.mkdtemp()`][].
+
+The optional `options` argument can be a string specifying an encoding, or an
+object with an `encoding` property specifying the character encoding to use.
```

Comment:
What about the callback-based version of this?

```js
fs.mkdirTempDisposable('prefix', (disposer) => {
  ??
});
```

I'm not convinced we should have it but we have sync/callback/promise based versions of all the methods so this would be a departure from the norm that should at least be discussed. I'm almost wondering if we shouldn't just have the one version of the method accessible via `fs.promises` and not worry about this sync function variation at all.

Reply:
Yeah, and that's fine. But given the asymmetry with everything else having a sync and callback variant we should document the difference here